### PR TITLE
Fix formatting regression

### PR DESCRIPTION
**Pull request by princemaple**
Bring formatting back on Windows. Regression in 0.6.0

### CODE REVIEW

File: apps/language_server/lib/language_server/providers/formatting.ex [hunk 1]
```diff
@@ -50,7 +50,8 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
   end
 
   def should_format?(file_uri, project_dir, inputs) when is_list(inputs) do
-    file = String.trim_leading(file_uri, "file://")
+    project_dir = project_dir |> String.downcase()
+    file_path = file_uri |> SourceFile.path_from_uri() |> String.downcase()
```
Comment (lukaszsamson) [inline apps/language_server/lib/language_server/providers/formatting.ex:54]: I'm afraid it won't work on case-sensitive file systems

Comment (princemaple) [reply apps/language_server/lib/language_server/providers/formatting.ex:54]: Hi @lukaszsamson , this is exactly what's happening going on in `can_format?` a few lines above. (I held off the refactor)
If there is a better way, I've happy to improve the PR and fix `can_format?` as well.

**Comment by axelson** [comment]
Do you get an error when formatting on windows?

**Comment by princemaple** [comment]
@axelson I simply can't format because the path can never match

File: apps/language_server/lib/language_server/providers/formatting.ex (hunk 1 above)
Comment (princemaple) [reply apps/language_server/lib/language_server/providers/formatting.ex:54]: That code has been there for 2 years, so I'm guessing it's fine? 🤷‍♂️

Comment (predragnikolic) [reply apps/language_server/lib/language_server/providers/formatting.ex:54]: @princemaple
I also think that this won't work for case-sensitive file systems.

Comment (lukaszsamson) [reply apps/language_server/lib/language_server/providers/formatting.ex:54]: > That code has been there for 2 years, so I'm guessing it's fine? 🤷‍♂️

Maybe it was broken the whole time. I can't check it right now, I usually work on CI filesystems (macOS and Windows).
Anyway it seems the only reliable way of determining if the file system is CI or CS is by creating/enumerating files, see e.g. https://stackoverflow.com/questions/430256/how-do-i-determine-whether-the-filesystem-is-case-sensitive-in-net/430810

Comment (princemaple) [reply apps/language_server/lib/language_server/providers/formatting.ex:54]: > I usually work on CI filesystems (macOS and Windows)

OK, same here.

**Comment by axelson** [comment]
@princemaple Can you give examples of the inputs that you're receiving for file_uri, project_dir, inputs?

**Comment by princemaple** [comment]
I'm not sure how to log in elixir-ls directly, so I dove into sublime-lsp and logged everything out there.
From the json rpc logs I found:

**rootUri**: `file:///C:/Users/chenp/code/project-name`
**projectDir**: `C:\\Users\\chenp\\code\\project-name`
**uri**: `file:///C:/Users/chenp/code/project-name/lib/app_web.ex`

These should be the raw inputs that elixir-ls received.

**Comment by princemaple** [comment]
FYI I set `projectDir` while testing different options. Previously it was `null`.

File: apps/language_server/lib/language_server/providers/formatting.ex (hunk 1 above)
Comment (princemaple) [reply apps/language_server/lib/language_server/providers/formatting.ex:54]: I've come to the conclusion that the bug is that `Path.wildcard` returns `c:/` in lowercase.

**Comment by princemaple** [comment]
I've come to the conclusion that the bug is that Path.wildcard returns c:/ in lowercase.

**Comment by princemaple** [comment]
Hi @lukaszsamson and @axelson I've sent https://github.com/elixir-lsp/elixir-ls/pull/375 
Please take a look.

**Comment by lukaszsamson** [comment]
> I've come to the conclusion that the bug is that `Path.wildcard` returns `c:/` in lowercase.

Should we report it upstream? IMO Path module functions shouldn't mess with casing even on CI filesystems.

**Comment by princemaple** [comment]
> Should we report it upstream? IMO Path module functions shouldn't mess with casing even on CI filesystems.

Maybe.
It's hard to say what is the **correct** case to output `C:/` in, since it's case insensitive 🤷‍♂️
It's not really "messing" with casing, it just happens to be reporting in lower case. And lower cases works, too, if you use it to list dir or check file existence.

The real question is, is there a better way to compare path than string comparison.
It's a solved problem in many environments, e.g.
image

**Comment by princemaple** [comment]
#375 replaces this one

**Pull request closed without merging**